libraries/extractor/src/main/java/androidx/media3/extractor/ts/TsExtractor.java

```diff
@@ -305,8 +305,9 @@ public void release() {
     long inputLength = input.getLength();
     if (tracksEnded) {
       boolean canReadDuration = inputLength != C.LENGTH_UNSET && mode != MODE_HLS;
-      if (canReadDuration && !durationReader.isDurationReadFinished()) {
-        return durationReader.readDuration(input, seekPosition, pcrPid);
+      int pesPid = getFirstPesReaderPid();
+      if (canReadDuration && !durationReader.isDurationReadFinished() && pesPid > 0) {
+        return durationReader.computeDuration(input, seekPosition, pesPid);
```

Comment:
Where is the fallback logic to use `readDuration` as before when DTS data is not available? Also why is it more accurate to use DTS data to calculate the duration rather than the PCR apart from the use case mentioned in the issue description?

Reply:
I've added the fallback in `readDuration`. The problem with the PCR is what I mentioned in the PR description, will elaborate more in there.